### ISSUE TITLE
HTTP FAILED: java.io.IOException: Canceled

### ISSUE THREAD

**Issue by Joadar** [square/okhttp#3276]
Hi,
First of all, I would like to thank you for the great job on it :)

Since the beginnings I implemented okhttp on my app, I get this kind of message in my logcat.
It's not a crash, just a debug message present in my logcat.

I get this one:
`04-08 18: 09: 45.385 8126-10529 /? D / OkHttp: <- HTTP FAILED: java.io.IOException: Canceled
`
And sometimes this one:
`HTTP FAILED: okhttp3.internal.http2.StreamResetException: stream was reset: CANCEL
`

Here are the things, I have a network call on a HTTPS API and the majority of my calls are fetch without problem but sometimes, I get these message and I do not understand why. I searched on Internet, I tried everything I found but I still don't fetch the data from the API for the current call.
I use Stetho and I can see on the network calls the same message instead of the result code.

Here my dependencies used:
```
// Retrofit
    Compile ("com.squareup.retrofit2: adapter-rxjava: 2.2.0") {
        Exclude group: 'com.squareup.okhttp3'
    }
    Compile ("com.squareup.retrofit2: converter-gson: 2.2.0") {
        Exclude group: 'com.squareup.okhttp3'
    }

 // Network
    Compile "com.squareup.okhttp3: logging-interceptor: 3.6.0"
```

I use for this app RxJava (1.2.3) and RxAndroid (1.2.1)

So why for the most of my network calls I successfully fetch data and for some I do not?
How can I retry the current call which failed?

Thanks in advance.

**Comment by digitalbuddha** [comment]
I believe what is happening is that you unsubscribe from an observable that was created by retrofit. When this happens retrofit will call cancel on the okhttp request. I had similar error when i rotated and unsubscribe from retrofit while a network stream was still being read.  
2 solutions I see are keeping the subscription alive or swallowing the exception. Your use case should dictate which works better for you.

**Comment by Joadar** [comment]
Hum, thanks for your answer but I don't get why this happening only for a minority of the network calls.

I display a list of item and when I click on one item, I load more data from API and once everytime, I get this network issue. It's not a different call (except parameters), I don't change anything during this call (rotation or something else), I don't get it :/

When I access to the item, I made two calls, once the first one is completed I made the second. If I put or don't put my Observable in a _CompositeSubscription_ I get the same issue. And my subscribe display data or display error to the user. But in any case the Observable goes to the onError method when the network call fail.
